recipe-dsl/src/main/scala/com/ing/baker/recipe/common/InteractionFailureStrategy.scala

```diff
@@ -1,30 +1,59 @@
 package com.ing.baker.recipe.common
 
-import scala.concurrent.duration.{Duration, FiniteDuration}
+import scala.annotation.tailrec
+import scala.concurrent.duration.Duration
 
 object InteractionFailureStrategy {
 
   case class RetryWithIncrementalBackoff(initialTimeout: Duration,
-                                         backoffFactor: Double = 2.0,
-                                         maximumRetries: Int = 50)
-      extends InteractionFailureStrategy {
+                                         backoffFactor: Double,
+                                         maximumRetries: Int,
+                                         maxTimeBetweenRetries: Option[Duration] = None)
+    extends InteractionFailureStrategy {
     require(backoffFactor >= 1.0, "backoff factor must be greater or equal to 1.0")
     require(maximumRetries >= 1, "maximum retries must be greater or equal to 1")
   }
 
+  @tailrec
+  private def determineHowManyTimesUntilMaxTimeIsMet(lastDelay: Duration,
+                                                     backoffFactor: Double,
+                                                     maxTimeBetweenRetries: Option[Duration],
+                                                     deadline: Duration,
+                                                     durationCounter: Duration,
+                                                     timesCounter: Int): Int = {
+    val nextDelay = determineNextDelay(lastDelay, backoffFactor, maxTimeBetweenRetries)
+    if ((durationCounter + nextDelay) > deadline) timesCounter
+    else determineHowManyTimesUntilMaxTimeIsMet(nextDelay, backoffFactor, maxTimeBetweenRetries, deadline, durationCounter + nextDelay, timesCounter + 1)
+  }
+
```

Comment:
I solved this using tail recursive since now the maxTimeBetweenRetries is added.
I did not know a better solution then to "brute force" the solution by just counting it.
Since this is only executed once it should not matter.

If someone knows a more elegant solution please implement it.